lib/core-integration/src/Test/Integration/Scenario/API/Shelley/Settings.hs

```diff
@@ -76,3 +81,40 @@ spec = describe "SHELLEY_SETTINGS" $ do
                     (`shouldBe` (either (const (error "no")) id $ fromText
                         @PoolMetadataSource uri))
                 ]
+
+    it "SETTINGS_02 - Changing pool_metadata_source re-syncs metadata" $ \ctx -> do
+        let toNone = "none"
+            toDirect = "direct"
+            getMetadata = map (view #metadata) . snd <$> unsafeRequest
+                @[ApiStakePool] ctx (Link.listStakePools arbitraryStake) Empty
+
+        updateMetadataSource ctx toNone
+        eventually "1. There is no metadata" $ do
+            metadatas <- getMetadata
+            metadatas `shouldSatisfy` (all isNothing)
```

Comment:
Haven't checked if it compiles, but a little golfing so we could have it on one line maybe:
```suggestion
        eventually "1. There is no metadata" $ getMetadata >>= (`shouldSatisfy` all isNothing)
```

Reply:
done.